File: backend/components/metiers/administration_publique/tests/guides/test_guides.js
// test_guides.js – Tests JS avancés pour la conformité des guides Threed
const fs = require('fs');
const path = require('path');

describe('Guides Threed', () => {
  const guidesDir = path.resolve(__dirname, '../guides');
  it('doit contenir les guides critiques', () => {
    const files = fs.readdirSync(guidesDir);
    expect(files).toContain('ACCESSIBILITY_GUIDE.md');
    expect(files).toContain('PLUGINS_GUIDE.md');
    expect(files).toContain('RGPD_GUIDE.md');
  });
  it('chaque guide doit avoir du contenu', () => {
    ['ACCESSIBILITY_GUIDE.md', 'PLUGINS_GUIDE.md', 'RGPD_GUIDE.md'].forEach(guide => {
      const content = fs.readFileSync(path.join(guidesDir, guide), 'utf-8');
      expect(content.length).toBeGreaterThan(10);
    });
  });
});

File: backend/components/metiers/administration_publique/tests/guides/test_guides.py
"""
Tests avancés pour la conformité des guides Threed (présence, contenu, accessibilité)
"""
import os
import pytest

def test_guides_presence():
    guides_dir = os.path.abspath(os.path.join(os.path.dirname(__file__), '../../guides'))
    guides = [f for f in os.listdir(guides_dir) if f.endswith('.md')]
    assert 'ACCESSIBILITY_GUIDE.md' in guides
    assert 'PLUGINS_GUIDE.md' in guides
    assert 'RGPD_GUIDE.md' in guides

def test_guides_content():
    guides_dir = os.path.abspath(os.path.join(os.path.dirname(__file__), '../../guides'))
    for guide in ['ACCESSIBILITY_GUIDE.md', 'PLUGINS_GUIDE.md', 'RGPD_GUIDE.md']:
        with open(os.path.join(guides_dir, guide)) as f:
            content = f.read()
            assert len(content) > 10
